docs/examples/write_read_dataset.rst

```diff
@@ -0,0 +1,67 @@
+Writing and reading a dataset using Lance
+---------------------------------------------------
+
+In this example, we will write a simple lance dataset to disk. Then we will read it and print out some basic properties like the schema and sizes for each record batch in the dataset.
+The example uses only one record batch, however it should work for larger datasets (multiple record batches) as well. 
+
+Writing the raw dataset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: rust
+
+    async fn write_dataset(data_path: &str) {
+
+        // Define new schema
+        let schema = Arc::new(Schema::new(vec![
+            Field::new("key", DataType::UInt32, false),
+            Field::new("value", DataType::UInt32, false),
+        ]));
+
+        // Create new record batch
+        let batch = RecordBatch::try_new(
+            schema.clone(),
+            vec![
+                Arc::new(UInt32Array::from(vec![1, 2, 3, 4, 5, 6])),
+                Arc::new(UInt32Array::from(vec![6, 7, 8, 9, 10, 11])),
+                ],
+        )
+        .unwrap();
+        
+        let batches = RecordBatchIterator::new([Ok(batch)], schema.clone());
+
+        // Define write parameters (e.g. overwrite dataset)
+        let write_params = WriteParams {
+            mode: WriteMode::Overwrite,
+            .. Default::default()
+        };
+
+        Dataset::write(batches, data_path, Some(write_params)).await.unwrap();
+    }
```

Comment:
Hmm, I don't think we want to copy/paste the example code.  Can you use a literalinclude instead?  There are some examples in `format.rst`